### PR TITLE
[PINOT-3248] First round check-in for re-factoring segment name code.

### PR DESCRIPTION
**Pull request by mcvsubbu**
Added classes that encapsulate segment names of all styles known so far.
Kept the old code for now, will remove once I can add tests to prove equivalence.

### CODE REVIEW

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java [hunk 1]
```diff
@@ -0,0 +1,136 @@
...
+    return _creationTime;
+  }
+
+  public String getSegmentName() {
+    return _segmentName;
+  }
+
+  @Override
+  public int compareTo(Object o) {
+    LLCSegmentNameHolder other = (LLCSegmentNameHolder)o;
```
Comment (jfim) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java]: I'd add a test for this, to make sure things are sorted in the expected order.

**Comment by jfim** [comment]
lgtm otherwise

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java (hunk 1 above)
Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java]: See SegmentNameBuilderTest.LLCHolderTest().

Does that suffice, or did you have more tests in mind?

**Comment by kishoreg** [comment]
Thanks for cleaning this. Is it possible to add a method that returns the type given segment name.
Lgtm

File: pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java (hunk 1 above)
Comment (jfim) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java]: I'd just add multiple segments in a sorted collection and/or do `Collections.sort` and then check in which order segments come out, which would document the sort order at the same time.

Comment (mcvsubbu) [inline pinot-common/src/main/java/com/linkedin/pinot/common/utils/LLCSegmentNameHolder.java]: Will do.

**Comment by mcvsubbu** [comment]
@kishoreg All in good time. I need to get this out now so that it unblocks us for continuing the low-level consumer work. Further re-factors will come soon to a PR near you :-)

**Pull request merged**